### PR TITLE
[PLAY-1477] Timestamp Kit Rails: Elapsed Time Variant Issue

### PR DESCRIPTION
**Pull request by skduncan**
**What does this PR do?** A clear and concise description with your runway ticket url.
https://runway.powerhrg.com/backlog_items/PLAY-1477

Creating private method in the PbTimestamp module to replace the "time_ago_in_words" ruby method to resolve issue in Nitro where the latter method conflicts with our Nitro behavior.

**Screenshots:** Screenshots to visualize your addition/change

**How to test?** Steps to confirm the desired behavior:
4. See addition/change

#### Checklist:
- [x] **LABELS** Add a label: `enhancement`, `bug`, `improvement`, `new kit`, `deprecated`, or `breaking`. See [Changelog & Labels](https://github.com/powerhome/playbook/wiki/Changelog-&-Labels) for details.
- [ ] **DEPLOY** I have added the `milano` label to show I'm ready for a review.
- [ ] **TESTS** I have added test coverage to my code.

### CODE REVIEW

File: playbook/app/pb_kits/playbook/pb_timestamp/timestamp.rb [hunk 1]
```diff
@@ -27,6 +27,15 @@ class Timestamp < Playbook::KitBase
...
                      default: "default"
 
+      # Variables to use with pb_time_ago method
+      SECS_PER_MIN  = 60
+      SECS_PER_HOUR = 60 * SECS_PER_MIN
+      SECS_PER_DAY  = 24 * SECS_PER_HOUR
+      SECS_PER_WEEK = 7 * SECS_PER_DAY
+      SECS_PER_MONTH = 4 * SECS_PER_WEEK
+      SECS_PER_YEAR = 12 * SECS_PER_MONTH
+      SECS_PER_CENT = 100 * SECS_PER_YEAR
```
Comment (kangaree) [inline playbook/app/pb_kits/playbook/pb_timestamp/timestamp.rb:37]: I was wondering if we'd ever show "centuries ago", but this is just the limit for time elapsed.

File: playbook/app/pb_kits/playbook/pb_timestamp/timestamp.rb [hunk 2]
```diff
@@ -73,12 +82,38 @@ def format_updated_string
...
+        datetime_string[1] = hide_updated ? datetime_string[1].upcase : datetime_string[1]
         updated_string = hide_updated ? "" : "Last updated"
-
         "#{updated_string}#{user_string}#{datetime_string}"
       end
 
+      def pb_time_ago(value)
+        time_ago = DateTime.now.to_i - value.to_i
+        case time_ago
+        when (0...SECS_PER_MIN)
```
Comment (kangaree) [inline playbook/app/pb_kits/playbook/pb_timestamp/timestamp.rb:94]: It makes sense to me to have 0-60 seconds for "seconds ago". But, this ends up being different from the React way: "0-44.99 seconds".

File: playbook/app/pb_kits/playbook/pb_timestamp/timestamp.rb [hunk 3]
```diff
@@ -73,12 +82,38 @@ def format_updated_string
...
+          time == 1 ? "a day" : "#{time_ago / SECS_PER_DAY} days"
+        when (SECS_PER_WEEK...SECS_PER_MONTH)
+          time = time_ago / SECS_PER_WEEK
+          time == 1 ? "a week" : "#{time_ago / SECS_PER_WEEK} weeks"
+        when (SECS_PER_MONTH...SECS_PER_YEAR)
+          time = time_ago / SECS_PER_MONTH
+          time == 1 ? "a month" : "#{time_ago / SECS_PER_MONTH} months"
+        when (SECS_PER_YEAR...SECS_PER_CENT)
+          time = time_ago / SECS_PER_YEAR
+          time == 1 ? "a year" : "#{time_ago / SECS_PER_YEAR} years"
```
Comment (kangaree) [inline playbook/app/pb_kits/playbook/pb_timestamp/timestamp.rb:113]: The React way is "320 days to 1+ year" (vs. 1+ year). 

Thinking a bit more, I kind of like the React way of doing things better. It'd be a bit weird to see a timestamp like "365 days ago" vs. "1 year ago".

File: playbook/spec/pb_kits/playbook/kits/timestamp_spec.rb [hunk 4]
```diff
@@ -128,7 +128,7 @@
         it "returns time elaspsed string including user's name" do
           timestamp = DateTime.new(2020, 10, 10, 20, 30, 0o0).in_time_zone("America/New_York").freeze
 
-          expect(subject.new(timestamp: timestamp, variant: variant, show_user: show_user, text: name).send(:format_elapsed_string)).to eq("Last updated by #{name} #{time_ago_in_words(timestamp)} ago")
+          expect(subject.new(timestamp: timestamp, variant: variant, show_user: show_user, text: name).send(:format_elapsed_string)).to eq("Last updated by #{name} #{subject.new.send(:pb_time_ago, timestamp)} ago")
```
Comment (kangaree) [inline playbook/spec/pb_kits/playbook/kits/timestamp_spec.rb:131]: Why `subject.new.send` instead of the function?

**Review by kangaree — CHANGES_REQUESTED** [review]
It looks really good so far! 

A few things, though:

- Tweak the ranges to match the React way. i.e., minutes ago is 90s-44.49 minutes, not 1minute-59.99minutes
- Update the rails doc example, https://pr3606.playbook.beta.gm.powerapp.cloud/kits/timestamp/rails#time-ago, to match React (10 hours ago)

File: playbook/spec/pb_kits/playbook/kits/timestamp_spec.rb (hunk 4 above)
Comment (skduncan) [reply playbook/spec/pb_kits/playbook/kits/timestamp_spec.rb:131]: Adding note for clarity. Due to how private functions work in rails, we can't just directly call them. They need to be called via the object/model they are located in.

**Review by kangaree — APPROVED** [review]

**Review by nidaqg — APPROVED** [review]

**Pull request merged**